src/Microsoft.OData.Core/UriParser/Parsers/UriQueryExpressionParser.cs

```diff
@@ -44,7 +44,7 @@ public sealed class UriQueryExpressionParser
         /// <summary>
         /// The current recursion depth.
         /// </summary>
-        private int recursionDepth;
+        internal int recursionDepth;

```

Comment:
```suggestion
        internal int RecursionDepth { get; set; }

```
nit: Maybe a property would be better for this.

Reply:
I refractored the code and reverted this change.